master/internal/api_project.go

```diff
@@ -342,3 +343,114 @@ func (a *apiServer) UnarchiveProject(
 	}
 	return &apiv1.UnarchiveProjectResponse{}, nil
 }
+
+func (a *apiServer) GetExperimentGroups(
+	ctx context.Context, req *apiv1.GetExperimentGroupsRequest) (*apiv1.GetExperimentGroupsResponse,
+	error,
+) {
+	resp := &apiv1.GetExperimentGroupsResponse{}
+	err := a.m.db.QueryProto("get_experiment_groups", &resp.Groups, req.ProjectId)
```

Comment:
As [Ilia mentioned in Slack](https://determined-ai.slack.com/archives/C6D7MDSQ5/p1664566350696609), we're moving toward using Bun for our queries and are trying to add no more `QueryProto` queries. The code here would look something like this:
```go
var groups ExperimentGroups
db.Bun().NewSelect().Model(&groups)
	Column("id").Column("name").Column("project_id").
	ColumnExpr("(SELECT COUNT(*) FROM experiments WHERE group_id = g.id) AS num_experiments").
	Where("g.project_id = ?", req.ProjectId).
	Scan(ctx)

return groups.Proto(), nil
```

We've also moved to putting these kinds of things in their own Go packages and such. For a more detailed example, I recommend checking out how we did User Groups in PRs #4519 and #4620. I'm happy to do a call or chat on Slack about this with you if you'd like!